pages/datasets/_datasetId.vue

```diff
@@ -294,24 +295,43 @@ const getThumbnailData = async (datasetDoi, datasetId, datasetVersion, datasetFa
             species = speciesArray[0].children[0].label.toLowerCase()
         }
         let taxo = Uberons.species['rat']
-        if (species && (species in Uberons.species))
+
+        if (species && species in Uberons.species)
           taxo = Uberons.species[species]
-          
-        scicrunchData.organs.forEach(organ => {
-          let organData = {
-            taxo,
-            uberonid: organ.curie,
-            organ: organ.name,
-            id: datasetId,
-            version: datasetVersion
+
+        // Check if flatmap has the anatomy for this species. This is done by asking the flatmap knowledge base
+        // if a flatmap of (species) has (anatomy)
+        let foundAnatomy = []
+        if (scicrunchData.organs[0]) { // Check if dataset has organ annotation
+          const anatomy = scicrunchData.organs.map(organ => organ.curie)
+          let data = await flatmaps.anatomyQuery(taxo, anatomy)
+          let anatomyResponse = data.data.values
+          if (anatomyResponse && anatomyResponse.length > 0) {
+            foundAnatomy = anatomyResponse.map(val => val[1]) // uberon is stored in second element of tuple
           }
-          flatmapData.push(organData)
+        }
 
-        });
+        // Add flatmaps that match the anatomy and taxonomy to the gallery
+        scicrunchData.organs.forEach(organ => {
+          if (foundAnatomy.includes(organ.curie)){
```

Comment:
Does the foundAnatomy contain other curies beside the one provided in the request?

Reply:
No, it only contains the ones found in the request. 

It is just there for readability and to handle the case where the flatmap query returns `[[]]`